exporter/opentelemetry-exporter-otlp/src/opentelemetry/exporter/otlp/metrics_exporter/__init__.py

```diff
@@ -167,37 +144,90 @@ def _translate_data(
                     sdk_metric.instrument.meter.resource
                 ] = InstrumentationLibraryMetrics()
 
-            self._metric_descriptor_kwargs = {}
+            type_class = {
+                int: {
+                    "sum": {"class": IntSum, "argument": "int_sum"},
+                    "gauge": {"class": IntGauge, "argument": "int_gauge"},
+                    "data_point_class": IntDataPoint,
+                },
+                float: {
+                    "sum": {"class": DoubleSum, "argument": "double_sum"},
+                    "gauge": {
+                        "class": DoubleGauge,
+                        "argument": "double_gauge",
+                    },
+                    "data_point_class": DoubleDataPoint,
+                },
+            }
+
+            value_type = sdk_metric.instrument.value_type
+
+            sum_class = type_class[value_type]["sum"]["class"]
+            gauge_class = type_class[value_type]["gauge"]["class"]
+            data_point_class = type_class[value_type]["data_point_class"]
+
+            if isinstance(sdk_metric.instrument, Counter):
+                otlp_metric_data = sum_class(
+                    data_points=_get_data_points(sdk_metric, data_point_class),
+                    aggregation_temporality=(
+                        AggregationTemporality.AGGREGATION_TEMPORALITY_DELTA
+                    ),
+                    is_monotonic=True,
+                )
+                argument = type_class[value_type]["sum"]["argument"]
 
-            metric_descriptor = MetricDescriptor(
-                name=sdk_metric.instrument.name,
-                description=sdk_metric.instrument.description,
-                unit=sdk_metric.instrument.unit,
-                type=_get_type(sdk_metric.instrument.value_type),
-                temporality=_get_temporality(sdk_metric.instrument),
-            )
+            elif isinstance(sdk_metric.instrument, UpDownCounter):
+                otlp_metric_data = sum_class(
+                    data_points=_get_data_points(sdk_metric, data_point_class),
+                    aggregation_temporality=(
+                        AggregationTemporality.AGGREGATION_TEMPORALITY_DELTA
+                    ),
+                    is_monotonic=False,
+                )
+                argument = type_class[value_type]["sum"]["argument"]
 
-            if metric_descriptor.type == MetricDescriptor.Type.INT64:
+            elif isinstance(sdk_metric.instrument, (ValueRecorder)):
+                logger.warning("Skipping exporting of ValueRecorder metric")
```

Comment:
Added an issue to track the work to support ValueRecorder: https://github.com/open-telemetry/opentelemetry-python/issues/1167